recipes/openssl/1.x.x/conanfile.py

```diff
@@ -223,35 +216,61 @@ def config_options(self):
 
     def configure(self):
         if self.options.shared:
-            del self.options.fPIC
-        del self.settings.compiler.libcxx
-        del self.settings.compiler.cppstd
-
-    def layout(self):
-        pass
+            self.options.rm_safe("fPIC")
+        self.settings.rm_safe("compiler.libcxx")
+        self.settings.rm_safe("compiler.cppstd")
+        self.conf.define("tools.gnu:make_program", self._make_program)
 
     def requirements(self):
-        if self._full_version < "1.1.0" and self.options.get_safe("no_zlib") == False:
+        if self._full_version < "1.1.0" and not self.options.get_safe("no_zlib"):
             self.requires("zlib/1.2.12")
 
     def validate(self):
-        if self.settings.os == "Emscripten":
+        if self.info.settings.os == "Emscripten":
```

Comment:
This advice has changed in recent weeks

```suggestion
        if self.settings.os == "Emscripten":
```